src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -52,10 +59,65 @@ class RegularSync(
       askForHeaders()
   }
 
-  def running: Receive = handleCommonMessages orElse {
-    case ResumeRegularSync =>
-      askForHeaders()
+  private def resumeRegularSync(): Unit = {
+    resumeRegularSyncTimeout.foreach(_.cancel)
+    resumeRegularSyncTimeout = None
+    askForHeaders()
+  }
 
+  /**
+    * Handles broadcasted blocks, processing them and inserting them to the blockchain if necessary
+    * FIXME: Current implementation is naive in that NewBlock msgs that can't be immediately processed are discarded
+    *        and requested again to the peers
+    */
+  def handleBroadcastedBlockMessages: Receive = {
+    case MessageFromPeer(NewBlock(newBlock, _), peerId) =>
+      //we allow inclusion of new block only if we are not syncing / reorganising chain
+      if (headersQueue.isEmpty && waitingForActor.isEmpty) {
+        val currentBestBlock = appStateStorage.getBestBlockNumber()
+        val maybeBlockParentTd = blockchain.getBlockHeaderByHash(newBlock.header.parentHash)
+          .flatMap(b => blockchain.getTotalDifficultyByHash(b.hash))
+        val maybeCurrentBlockTd = blockchain.getTotalDifficultyByNumber(currentBestBlock)
+
+        (maybeBlockParentTd, maybeCurrentBlockTd) match {
+          case (_, None) =>
+            //Something went wrong, we don't have the total difficulty of the latest block
+            //TODO: Investigate if we can recover from this error [EC-165]
+            log.error(s"No total difficulty for the latest block with number $currentBestBlock")
+            context stop self
```

Comment:
maybe we should close the whole app then? is there a point of having the process running when it's no longer syncing?